### PR TITLE
sioyek: unbreak on darwin

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

This pr can build and run sioyek on darwin successfully.

Reference: https://github.com/ahrm/sioyek/blob/main/build_mac.sh

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/sioyek/default.nix [hunk 1]
```diff
@@ -25,29 +26,44 @@ stdenv.mkDerivation rec {
...
   postPatch = ''
+    substituteInPlace pdf_viewer_build_config.pro \
+      --replace "-lmupdf-threads" "-lgumbo -lharfbuzz -lfreetype -ljbig2dec -ljpeg -lopenjp2"
     substituteInPlace pdf_viewer/main.cpp \
       --replace "/usr/share/sioyek" "$out/share" \
       --replace "/etc/sioyek" "$out/etc"
   '';
 
-  postInstall = ''
+  postInstall = if stdenv.isLinux then ''
```
Comment (risicle) [inline pkgs/applications/misc/sioyek/default.nix:44]: This assumes that anything that's not `isLinux` is darwin and goes ahead and installs macos-specific stuff. I don't think we can make that assumption and it's much better to activate the macos-specific stuff when we explicitly detect darwin.

Comment (jiegec) [reply pkgs/applications/misc/sioyek/default.nix:44]: Done

Comment (risicle) [reply pkgs/applications/misc/sioyek/default.nix:44]: This'll work, but I think what I meant was that darwin should have the special case, and all the other paths should follow the "linux" approach. Other platforms are likely to be much more like linux than macos.

Comment (jiegec) [reply pkgs/applications/misc/sioyek/default.nix]: I see, updated

**Review by risicle — APPROVED** [review]
WFM, macos 10.15. Builds happily, nixos x86_64.

**Pull request merged**